### PR TITLE
Spec v2/fix/misc fixes

### PR DESCRIPTION
**Pull request by shreyaxh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/dev/migrations/dynamic-offer-driver-app/0365-beckn-config-mobility.sql [hunk 1]
```diff
@@ -26,7 +27,7 @@ VALUES
         'MOBILITY',
         'http://localhost:8015/v1',
         'http://localhost:8020',
-        'localhost:8013/beckn/cab/v1/da4e23a5-3ce6-4c37-8b9b-41377c3c1a52',
+        'NAMMA_YATRI',
```
Comment (NikithShetty) [inline Backend/dev/migrations/dynamic-offer-driver-app/0365-beckn-config-mobility.sql:30]: you can keep this the same as before.

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs [hunk 2]
```diff
@@ -114,8 +115,8 @@ tfItems res =
 -- TODO: Discuss payment info transmission with ONDC
 tfPayments :: DOnInit.OnInitRes -> DBC.BecknConfig -> Maybe [Spec.Payment]
 tfPayments res bapConfig = do
-  let amount = fromIntegral (res.estimatedTotalFare.getMoney)
-  let mkParams :: (Maybe BknPaymentParams) = (readMaybe . T.unpack) =<< bapConfig.paymentParamsJson
+  let amount = HighPrecMoney (fromMaybe 0.0 $ A.decode $ A.encode res.estimatedTotalFare.getMoney)
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs:118]: Don't default to `0.0`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs [hunk 3]
```diff
@@ -687,14 +688,21 @@ mapVariantToVehicle variant = do
...
+mapRideStatus rideStatus = do
+  case rideStatus of
+    DRide.NEW -> Enums.RIDE_ASSIGNED
+    DRide.INPROGRESS -> Enums.RIDE_STARTED
+    DRide.COMPLETED -> Enums.RIDE_ENDED
+    DRide.CANCELLED -> Enums.RIDE_CANCELLED
+
+tfCancellationFee :: Maybe Int -> Maybe Spec.Fee
+tfCancellationFee mbAmount = do
   let amount = fromMaybe 0 mbAmount
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs:701]: Don't default to `0`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnInit.hs [hunk 4]
```diff
@@ -67,18 +72,17 @@ tfFulfillments res =
 -- TODO: Discuss payment info transmission with ONDC
 tfPayments :: DInit.InitRes -> DBC.BecknConfig -> Maybe [Spec.Payment]
 tfPayments res bppConfig = do
-  let amount = fromIntegral (res.booking.estimatedFare.getMoney)
-  let mkParams :: (Maybe BknPaymentParams) = (readMaybe . T.unpack) =<< bppConfig.paymentParamsJson
+  let amount = HighPrecMoney (fromMaybe 0.0 $ A.decode $ A.encode res.booking.estimatedFare.getMoney)
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnInit.hs:75]: Remove defaults here too

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 5]
```diff
@@ -84,8 +90,8 @@ tfFulfillments res = do
 -- TODO: Discuss payment info transmission with ONDC
 tfPayments :: DConfirm.DConfirmResp -> DBC.BecknConfig -> Maybe [Spec.Payment]
 tfPayments res bppConfig = do
-  let amount = fromIntegral (res.booking.estimatedFare.getMoney)
-  let mkParams :: (Maybe BknPaymentParams) = (readMaybe . T.unpack) =<< bppConfig.paymentParamsJson
+  let amount = HighPrecMoney (fromMaybe 0.0 $ A.decode $ A.encode res.booking.estimatedFare.getMoney)
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:93]: Remove defaults here too

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs [hunk 6]
```diff
@@ -804,8 +813,11 @@ mkQuotationBreakup booking =
 
 type MerchantShortId = Text
 
-tfItems :: DBooking.Booking -> MerchantShortId -> Maybe Pricing -> Maybe [Spec.Item]
-tfItems booking shortId pricing =
+tfItems :: DBooking.Booking -> MerchantShortId -> Maybe Meters -> Maybe FarePolicyD.FarePolicy -> Maybe [Spec.Item]
+tfItems booking shortId estimatedDistance mbFarePolicy = do
+  let itemTags = case mbFarePolicy of
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs:818]: we can use `fmap` here

File: Backend/lib/beckn-spec/src/BecknV2/OnDemand/Utils/Common.hs [hunk 7]
```diff
@@ -48,3 +50,6 @@
       case A.eitherDecodeStrict reqBS of
         Right reqV2 -> pure $ Right reqV2
         Left err -> throwError . InvalidRequest $ "Unable to parse request: " <> T.pack err <> T.decodeUtf8 reqBS
+
+gpsToText :: Gps.Gps -> Maybe Text
+gpsToText Gps.Gps {..} = Just $ (T.pack $ printf "%.6f" lat) <> ", " <> (T.pack $ printf "%.6f" lon)
```
[the remark this answers is not in the document]
Comment (NikithShetty) [reply Backend/lib/beckn-spec/src/BecknV2/OnDemand/Utils/Common.hs:55]: Fix hlint warning

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/OnDemand/Utils/OnSearch.hs [hunk 8]
```diff
@@ -231,3 +231,8 @@ makeLatLong location = do
 buildSpecialLocationTag :: MonadFlow m => Spec.Item -> m (Maybe Text)
 buildSpecialLocationTag item =
   return $ Utils.getTagV2 Tag.INFO Tag.SPECIAL_LOCATION_TAG item.itemTags
+
+getDescriptorInfo :: MonadFlow m => Spec.Item -> m (Maybe Text)
+getDescriptorInfo item = do
+  let serviceTierName = item.itemDescriptor >>= (.descriptorName)
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/OnDemand/Utils/OnSearch.hs:237]: could be one liner

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSearch.hs [hunk 9]
```diff
@@ -320,6 +322,7 @@ buildQuote requestId providerInfo now searchRequest QuoteInfo {..} = do
         quoteDetails = quoteDetails',
         merchantId = searchRequest.merchantId,
         merchantOperatingCityId = searchRequest.merchantOperatingCityId,
+        serviceTierName = Nothing,
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSearch.hs:325]: Why is `serviceTierName = Nothing` for Quote?

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSearch.hs (hunk 9 above)
Comment (jaypalmudaliyar24) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSearch.hs]: Fixed.

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**